esp/src/src-react/components/Title.tsx

```diff
@@ -236,16 +245,23 @@ export const DevTitle: React.FunctionComponent<DevTitleProps> = ({
         if (!cookie("PasswordExpiredCheck")) {
             // cookie expires option expects whole number of days, use a decimal < 1 for hours
             cookie("PasswordExpiredCheck", "true", { expires: 0.5, path: "/" });
-            if (currentUser.passwordIsExpired) {
-                alert(nlsHPCC.PasswordExpired);
-                setShowMyAccount(true);
-            } else if (currentUser.passwordDaysRemaining && currentUser.passwordDaysRemaining <= currentUser.passwordExpirationWarningDays) {
-                if (confirm(nlsHPCC.PasswordExpirePrefix + currentUser.passwordDaysRemaining + nlsHPCC.PasswordExpirePostfix)) {
-                    setShowMyAccount(true);
-                }
+            switch (currentUser.passwordDaysRemaining) {
+                case -1:
+                    setPasswordExpiredConfirm(true);
+                    break;
+                case -2:
+                case null:
+                    break;
```

Comment:
Should these conditions not display something?  Does -2, -3, -4 not also mean that the password is expired?

Reply:
ESP has two magic numbers for password lifespan, -1 implies that it has expired and -2 that it never expires.